flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -252,23 +261,31 @@ protected StreamTask(
 	 * Runs the stream-tasks main processing loop.
 	 */
 	private void run() throws Exception {
-		final ActionContext actionContext = new ActionContext();
+		final Mailbox mailbox = taskMailboxExecutor.getMailbox();
```

Comment:
Same as for `SourceStreamTask.runAlternativeMailboxLoop()`, you can add a precondition check that this method is called in the proper thread.